### ISSUE TITLE
[Bug]: Cross-posted to: same community multiple times

### ISSUE THREAD

**Issue by testman42** [LemmyNet/lemmy-ui#1243]
### Summary

same community can be listed multiple times in the list of where link was cross-posted to.
Example: https://lemmy.ml/post/1240605
Notice the duplicate mention of `gaming` here:
`cross-posted to: gaming gaming spiele@feddit.de`

### Steps to Reproduce

I assume it goes like this:
1.  find post that has some link
2. make new post to the same link in that same community 
3. notice the cross-posted list

### Technical Details

Does lemmy not filter the list to unique comminutes?
Is that actually intended behaviour? 
So that there is not just first post of it's kind, but that all duplicate posts are equally represented?

### Version

0.17.4

**Comment by Nutomic** [comment]
The crosspost check is very basic, it just gets a list of posts which have the same url. The display is misleading in that it shows community names when the links actually go to individual posts.

**Comment by alectrocute** [comment]
This was fixed by @dessalines in https://github.com/LemmyNet/lemmy/pull/3234.

**Comment by poVoq** [comment]
Can we re-open this as it isn't fixed in 0.18.4?

In fact I think it got worse, as now cross-posting it into the same community hides it from view in the main community listing on the web-ui.

See this post as an example: https://slrpnk.net/post/1747064

Somehow it ended up being cross-posted to the same community and if you go to https://slrpnk.net/c/meta it is hidden. However on the mlmym alternative UI which does not support cross-posts it can be seen: https://old.slrpnk.net/c/meta (only once though for some reason I fail to understand).

**Comment by dessalines** [comment]
Okay I see what's going on, and I'm not sure that its a bug. 

- 2 posts have the same link (so the cross-post should remove one of them)
- One post got posted a month ago, and it has a score of -16, and was the first one. It correctly shows the cross-posted to data: 
![Screenshot_2023-08-31-10-56-09-758_mark via gp](https://github.com/LemmyNet/lemmy-ui/assets/930215/acd91753-2d66-4a04-a3bd-5cd0ba56a64a)

The only thing I could maybe see to alter this, would be to make the *latter* post, or maybe the post with a higher score, be the most prominent one, rather than the oldest post. Either way you should probably open a new issue for this.

**Comment by poVoq** [comment]
Hmm, yeah makes sense... however for primarily discussion posts this is a bit unexpected behaviour.

I'll open a new one. Thanks for looking into it.
